src/adapters/methods/reft.py

```diff
@@ -26,6 +26,8 @@ def __init__(
 
         projection = nn.Linear(in_dim, r_dim, bias=False, dtype=dtype)
         if orthogonal:
+            # orthogonal is not implemented for half precision
+            projection = projection.to(dtype=torch.float32)
```

Comment:
2 things about explicitly converting to float32 here:
1. I believe we should warn users when we silently change their specified dtype, as this could lead to unexpected behavior or confusion. 
2. The current change forces float32 even when the user specifies other valid dtypes like float64. I don't know why anyone should choose other types, but it doesn't hurt to support this.

Maybe changing it to these lines would be better:
```suggestion
            if dtype in [torch.float16, torch.bfloat16]:
                warnings.warn(
                    "Orthogonal parametrization is not supported for half precision dtypes. Converting REFT projection layer to float32.",
                    UserWarning
                )
                projection = projection.to(dtype=torch.float32)
```